### ISSUE TITLE
chromecast audio not recognized?

### ISSUE THREAD

**Issue by molberg** [masmu/pulseaudio-dlna#107]
Hi,
The changelog lists support for chromecast audio for version 0.4.5.2. I installed this version via the ppa. But on start pulseaudio-dlna outputs:

"The Chromecast seems not to be an original Chromecast! Model name: "Chromecast Audio" Skipping device ..."

and it does not show up in pulseaudio control. Am I doing something wrong?

**Comment by masmu** [comment]
It list support in _master_. Want to build a release pretty soon. Just some problems left to solve.

**Comment by masmu** [comment]
Btw, do you also own a _Chromecast_ 2?

**Comment by molberg** [comment]
yes, bought both of them today ;-). Support in Linux was a major driver for me.

**Comment by molberg** [comment]
Take your time, I really appreciate the work you are doing. Let me know if I can help testing things.

**Comment by masmu** [comment]
Thats great. I expect the new chromecast 2015 (HDMI) model to also work. But I need the _device-desc.xml_ the device provides, to put the device's _friendlyName_ on the white list. Since i am just owning the old Chromecast and the new Chromecast Audio this is a problem for me :wink: 

Could you paste that here?

```bash
wget http://<chromecast-ip>:8008/ssdp/device-desc.xml
```

**Comment by molberg** [comment]
<?xml version="1.0"?>
<root xmlns="urn:schemas-upnp-org:device-1-0">
  <specVersion>
    <major>1</major>
    <minor>0</minor>
  </specVersion>
  <URLBase>http://192.168.1.83:8008</URLBase>
  <device>
    <deviceType>urn:dial-multiscreen-org:device:dial:1</deviceType>
    <friendlyName>TV-rummet (video)</friendlyName>
    <manufacturer>Google Inc.</manufacturer>
    <modelName>Eureka Dongle</modelName>
    <UDN>uuid:80d0d4bf-50ec-52e5-a973-240b4bcb0ffd</UDN>
    <iconList>
      <icon>
        <mimetype>image/png</mimetype>
        <width>98</width>
        <height>55</height>
        <depth>32</depth>
        <url>/setup/icon.png</url>
      </icon>
    </iconList>
    <serviceList>
      <service>
        <serviceType>urn:dial-multiscreen-org:service:dial:1</serviceType>
        <serviceId>urn:dial-multiscreen-org:serviceId:dial</serviceId>
        <controlURL>/ssdp/notfound</controlURL>
        <eventSubURL>/ssdp/notfound</eventSubURL>
        <SCPDURL>http://www.google.com/cast</SCPDURL>
      </service>
    </serviceList>
  </device>
</root>

**Comment by molberg** [comment]
I'll try again with quotes:

```xml
<?xml version="1.0"?>
<root xmlns="urn:schemas-upnp-org:device-1-0">
  <specVersion>
    <major>1</major>
    <minor>0</minor>
  </specVersion>
  <URLBase>http://192.168.1.83:8008</URLBase>
  <device>
    <deviceType>urn:dial-multiscreen-org:device:dial:1</deviceType>
    <friendlyName>TV-rummet (video)</friendlyName>
    <manufacturer>Google Inc.</manufacturer>
    <modelName>Eureka Dongle</modelName>
    <UDN>uuid:80d0d4bf-50ec-52e5-a973-240b4bcb0ffd</UDN>
    <iconList>
      <icon>
        <mimetype>image/png</mimetype>
        <width>98</width>
        <height>55</height>
        <depth>32</depth>
        <url>/setup/icon.png</url>
      </icon>
    </iconList>
    <serviceList>
      <service>
        <serviceType>urn:dial-multiscreen-org:service:dial:1</serviceType>
        <serviceId>urn:dial-multiscreen-org:serviceId:dial</serviceId>
        <controlURL>/ssdp/notfound</controlURL>
        <eventSubURL>/ssdp/notfound</eventSubURL>
        <SCPDURL>http://www.google.com/cast</SCPDURL>
      </service>
    </serviceList>
  </device>
</root>
```

**Comment by masmu** [comment]
That is from a new Chromecast 2 (HDMI)?

**Comment by molberg** [comment]
the "hockey puck" with hdmi output, yes.

**Comment by masmu** [comment]
Hmm, okay. Thanks, it identifies itself as the old Chromecast did. _Eureka Dongle_. I would have expected that there is a difference like the Chromecast Audio does. So, I have nothing to do.

Okay, It would be great if you could test the latest [code](https://github.com/masmu/pulseaudio-dlna/tree/feature/request-timeout). If you need instructions, don't hesitate to ask.

**Comment by molberg** [comment]
Works like a charm on both my devices! There is not supposed to be a picture on the tv when I am playing audio, is there? I see a grey frame with a blue line at the bottom edge. I sent you some money earlier, you earned it! Thanks!

**Comment by masmu** [comment]
Thank you!

Yes, the grey picture is intended. Was thinking about placing an application logo there. But since you can stream multiple applications to the same device it is a little bit trick to place coverart there.

**Comment by masmu** [comment]
Just waiting for some final test reports of other users, then it is time for _0.4.6_ ...

**Comment by chmduquesne** [comment]
Which kind of test report are you waiting for? I just bought a chromecast audio, let me know if I can help!

**Comment by masmu** [comment]
I added code for adjusting the [request timeout](https://github.com/masmu/pulseaudio-dlna/tree/feature/request-timeout) of HTTP requests. But the [guy](https://github.com/masmu/pulseaudio-dlna/issues/102) who requested this is not responding currently.

The code for this is not related to the Chromecast, but it would be nice if this could be in the next release. I don't wanna stress the package maintainers for other distributions too often to build new packages.

So, basically I am waiting for him to confirm that my changes solved his problem.

**Comment by masmu** [comment]
Packages are currently building. They should be available in 30 minutes via `apt-get`.

 * __0.4.6__ - (_2015-10-17_)
    - Added support for _Google Chromecast Audio_ (thanks to [leonhandreke](https://github.com/leonhandreke))
    - Fixed a bug where devices which does not specifiy control urls made the application crash
    - Added the `--disable-device-stop` flag
    - Added the `--request-timeout` option
    - You can now also add rules to renderers (e.g. `DISABLE_DEVICE_STOP`, `REQUEST_TIMEOUT`)
    - Fixed a bug where stream urls where not parsed correctly
    - Fixed a bug which made a Chomecast Audio throwing exceptions while stopping
    - Fixed a bug where the system's default encoding could not be determined when piping the applications output

**Comment by chmduquesne** [comment]
This is awesome. Thank you for this package!

**Comment by masmu** [comment]
You are welcome!

**Comment by masmu** [comment]
From now there should not be a grey picture any more ...

 * __0.4.7__ - (_2015-11-18_)
    - The application can now co-exist with other applications which are using the port 1900/udp (thanks to [klaernie](https://github.com/klaernie))
    - Fixed the daemon mode to support `psutil` 1.x and 2.x (thanks to [klaernie](https://github.com/klaernie))
    - HTML entities in device descriptions are now converted automatically
    - Faster and more reliable device discovery (new dependency `python-netifaces`)
    - Added the `--cover-mode` option, one mode requires (optional) dependencies `gtk`, `cairo`, `rsvg`
    - L16 codecs are now selected better (e.g. needed for _XBox 360_)
    - Fixed a bug where sometimes it was tried to remove sinks twice on cleanup
    - Added the `--update-device-config` flag
    - Added the `--ssdp-ttl`, `--ssdp-mx`, `--ssdp-amount` options
    - Added the `--msearch-port` option
